lib/liquid/partial_cache.rb

```diff
@@ -11,7 +11,10 @@ def self.load(template_name, context:, parse_context:)
       source = file_system.read_template_file(template_name)
       parse_context.partial = true
 
-      partial = Liquid::Template.parse(source, parse_context)
+      liquid_template_factory = (context.registers[:liquid_template_factory] ||= Liquid::TemplateFactory)
```

Comment:
You may need to modify this if this moves forward. `context.registers` are dropped when `context` is a `StaticRegister` and you enter a subcontext eg. `render` tags.

If you want this to flow down through subcontexts you will need to use some form of `context.registers.static[:liquid_template_factory]` however `static` only exists on `StaticRegisters`